Automattic/wp-calypso
# Hosting: Add visibility of last good backup

#### Changes proposed in this Pull Request

Adds a new "Site backup" card to the hosting section that displays the last good backup of the site.

![image](https://user-images.githubusercontent.com/1233880/69642593-45ac9880-1062-11ea-9c2a-2aebfcfd57c2.png)

To support this, we have added new actions, selectors and reducers to the rewind state. This has required to move the previous rewind state one level deeper, so backups can be stored in the same object:

**Before**
```
console.log( state.rewind );
```
Output:
``` 
{ 
  [siteId]: { 
    state: ..., 
    lastUpdated: ..., 
    ... 
  }
}
```

**After**
```
console.log( state.rewind );
```
Output:
``` 
{ 
  [siteId]: { 
    state: {
      state: ..., 
      lastUpdated: ..., 
      ... 
    },
    backups: [ ... ],
  }
}
```

#### Testing instructions

* Switch to a new Atomic site.
* Go to Manage > Hosting Configuration.
* Verify the Site backup card displays the latest good backup (it should match with what's shown in the activity log).
* Switch to an Atomic site which has not been backed up yet (i.e. a fresh new site).
* Verify the card displays a "There are no recent backups for your site" message.

Fixes #37258
